### ISSUE TITLE
error compiling on gcc 5.4

### ISSUE THREAD

**Issue by baaa** [fireice-uk/xmr-stak-cpu#173]
```ubuntu@xxxx:~/xmr-stak-cpu/xmr-stak-cpu-1.2.0-1.4.1$ cmake .
CMake Error at CMakeLists.txt:27 (message):
  GCC version must be at least 5.1!


-- Configuring incomplete, errors occurred!
See also "/home/ubuntu/xmr-stak-cpu/xmr-stak-cpu-1.2.0-1.4.1/CMakeFiles/CMakeOutput.log".
ubuntu@xxxx:~/xmr-stak-cpu/xmr-stak-cpu-1.2.0-1.4.1$ which gcc
/usr/bin/gcc
ubuntu@xxxx:~/xmr-stak-cpu/xmr-stak-cpu-1.2.0-1.4.1$ gcc --version
gcc (Ubuntu 5.4.1-2ubuntu1~14.04) 5.4.1 20160904
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

**Comment by psychocrypt** [comment]
please post the full cmake output

**Comment by psychocrypt** [comment]
please also post the output of `c++ --version`

**Comment by baaa** [comment]
i already terminated the instance, what do you mean all output? this is everything.

**Comment by psychocrypt** [comment]
all lines because the importend information is ehich compiler where found by cmake. I think your c+q compiler was to old and you checked only the c compiler

**Comment by baaa** [comment]
its rather easy to recreate, i'll try it again after my current setups finish (just for science!)
``` 
sudo add-apt-repository ppa:ubuntu-toolchain-r/test
sudo apt-get update
sudo apt-get install gcc-5 g++-5
sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-5 60 --slave /usr/bin/g++ g++ /usr/bin/g++-5
sudo apt-get install cmake3
```

download the tar, go to directory and cmake .

again like i said, the eror message i posted starts from line:1 nothing before that but me issusing cmake .

**Comment by uaktags** [comment]
I was able to reprod it on ubuntu 14.04 as well. Strangely first time it worked (but I forgot microhttpd) so I did a copy&pasta of the install deps from the front page, and then it failed during the subsequent cmake .

First run:
```
root@localhost:~/xmr-stak-cpu-1.2.0-1.4.1# cmake .
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 5.4.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
CMake Error at CMakeLists.txt:55 (message):
  microhttpd NOT found: use `-DMICROHTTPD_REQUIRED=OFF` to build without http
  deamon support


-- Configuring incomplete, errors occurred!
```

2nd time around after installing deps
```
root@localhost:~/xmr-stak-cpu-1.2.0-1.4.1# cmake .
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:27 (message):
  GCC version must be at least 5.1!

root@localhost:~/xmr-stak-cpu-1.2.0-1.4.1# gcc-5 --version
gcc-5 (Ubuntu 5.4.1-2ubuntu1~14.04) 5.4.1 20160904

root@localhost:~/xmr-stak-cpu-1.2.0-1.4.1# gcc --version
gcc (Ubuntu 5.4.1-2ubuntu1~14.04) 5.4.1 20160904
```
I can't explain the issue but I was able to get it after removing/purging build-essentials, gcc-5, g++-5,  reinstalling them and update-alternatives, and finally a logout/in. My only guess is that the Environment Variables got messed up as that's what CMake appears to be checking, but I could be dead wrong.
```
root@localhost:~/xmr-stak-cpu-1.2.0-1.4.1# cmake .
-- The C compiler identification is GNU 5.4.1
-- The CXX compiler identification is GNU 5.4.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.0.1f")
-- Configuring done
-- Generating done
-- Build files have been written to: /root/xmr-stak-cpu-1.2.0-1.4.1
```

**Comment by baaa** [comment]
that looks like the case...
looks like any apt-get install messes up "some variable" and cmake stops working. so the last thing installed has to be gcc g++ and it works.

**Comment by psychocrypt** [comment]
This issue is solved. 
- g++ and gcc needs to be available in version 5.1+
- for older linux version with the default compiler 4.X and a second g++/gcc 5.X the newer compiler must be chosen. This can also be done via environment variable e.g. `export CXX=g++-5` and `export CC=gcc-5`
